src/psyclone/tests/psyir/nodes/reference_test.py

```diff
@@ -256,6 +256,24 @@ def test_reference_next_accesses(fortran_reader):
     assert len(b.next_accesses()) == 1
     assert b.next_accesses()[0] == b
 
+    # Check that a loop accessing a variable before
+    # the reference doesn't result in a false positive.
+    code = '''subroutine my_sub()
+    integer :: a
+    integer :: b
+    do a = 0, 10
+        b = a
+    end do
+    a = 1
+    end subroutine'''
+    psyir = fortran_reader.psyir_from_source(code)
+    routine = psyir.children[0]
+    a = routine.children[1].lhs
```

Comment:
For clarity maybe call it "a_after_loop" and a_2 call it "a_inside_loop"

Reply:
Done

Reply:
Actually this a_2 was never used so I've now removed it.